packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -487,6 +487,23 @@ class AnimationController extends Animation<double>
     return _animateToInternal(target, duration: duration, curve: curve);
   }
 
+  /// Drives the animation from its current value to target.
```

Comment:
Reading this file in order. So far I'm a bit skeptical about adding new APIs at this low level since it seems like it's the same as [reverse] and the curve can come from the Animation being driven.

Reply:
So, I mean, the core of the issue is that it is necessary that the animation direction is ```reverse```, and there's no way to do that using a custom animation with the functions that we currently have and the way the whole animation system is set up for the route, at least as far as I could tell. Tho maybe I'm missing something?

Reply:
Talked offline. This is ok since we're already driving animation controllers non-linearly via calls to .fling etc.